src/app.jsx

```diff
@@ -467,3 +475,39 @@ const ForceDeleteModal = ({ selected, itemPath, errorMessage, deleteFailed }) =>
         </Modal>
     );
 };
+
+const CreateDirectoryModal = ({ currentPath, errorMessage }) => {
+    const Dialogs = useDialogs();
+    const [name, setName] = useState("");
+
+    const createDirectory = () => {
+        cockpit.spawn(["mkdir", currentPath + name], { superuser: "try" })
+                .then(Dialogs.close, (err) => { Dialogs.show(<CreateDirectoryModal currentPath={currentPath} errorMessage={err.message} />) });
+    };
+
+    return (
+        <Modal
+            position="top"
+            title={_("Create directory")}
+            isOpen
+            onClose={Dialogs.close}
+            footer={errorMessage === undefined &&
+                <>
+                    <Button variant='primary' onClick={createDirectory}>{_("Create")}</Button>
+                    <Button variant='link' onClick={Dialogs.close}>{_("Cancel")}</Button>
+                </>}
+        >
+            {errorMessage === undefined &&
+                <>
```

Comment:
You need to use Form element from patternfly.

Reply:
And then FormGroup for displaying the label, field.

Reply:
Plus you should always show the form contents, now only when there is no error.

Reply:
Updated! The error could use some better spacing though
![image](https://github.com/KKoukiou/cockpit-navigator/assets/90795679/83c7cbde-5f35-441e-ae5c-5754873a1abf)

Reply:
The error should appear on top of the form. You can use Stack layout for spacing